SqlReplay.Console/EventExecutor.cs

```diff
@@ -394,12 +396,54 @@ private void SetupSqlCommandParameters(SqlCommand cmd, Rpc rpc)
                 }
                 else
                 {
-                    sqlParam.Value = param.Value;
+                    sqlParam.Value = SqlTypifyValue(param.Value, param.SqlDbType);
                 }
                 cmd.Parameters.Add(sqlParam);
             }
         }
 
+        /// <summary>
+        /// Converts special case input into appropriate SQL values. See Remarks.
+        /// </summary>
+        /// <param name="inputValue">The unconverted parameter value to send to SQL</param>
+        /// <param name="sqlDbType">The SQL Type of the parameter</param>
+        /// <returns>A special default value if <paramref name="inputValue"/> is "" and <paramref name="sqlDbType"/> is
+        /// SqlDbType.Int or DateTime; otherwise, <paramref name="inputValue"/></returns>
+        /// <remarks>
+        /// <para>This oddball little method is required because some of our executions pass '' (empty string) to sproc
+        /// parameters of type INT or DATETIME. In SQL-Replay, though, ADO.Net is intercepting those executions
+        /// and complaining that "" doesn't convert to an Int32.
+        /// To limit risk exposure, this method is very tightly scoped to address only these two known occurring
+        /// problems.</para>
+        /// <para>Try this fun SQL out for a demo of what kind of type wrangling SQL allows when .NET isn't
+        /// mediating:</para>
+        /// <code>
+        /// DECLARE @defaultInt INT, @emptyStringInt INT = '', @oneSpaceInt INT = ' ', @twoSpaceInt INT = '  '
+        /// DECLARE @defaultdt DATETIME, @emptyStringDt DATETIME = '', @oneSpaceDt DATETIME = ' ', @twoSpaceDt DATETIME = '  '
+        /// SELECT @defaultInt, @emptyStringInt, @oneSpaceInt, @twoSpaceInt
+        /// SELECT @defaultdt, @emptyStringDt, @oneSpaceDt, @twoSpaceDt
+        /// -- FYI conversion fails for all the following examples:
+        /// DECLARE @tabInt INT = '	'
+        /// DECLARE @tabDt DATETIME = '	'
+        /// DECLARE @newlineInt INT = '
+        /// '
+        /// DECLARE @newlineDt DATETIME = '
+        /// '
+        /// DECLARE @alphaInt INT = 'abc'
+        /// DECLARE @alphaDt DATETIME = 'abc'
+        /// </code>
+        /// </remarks>
+        private static object SqlTypifyValue(object inputValue, SqlDbType sqlDbType)
+        {
+            return sqlDbType switch
+            {
+                SqlDbType.Int when inputValue is "" => 0,
```

Comment:
In your example above you mention there can be any number of spaces in the input string and it translates to zero. Does this handle those cases or are you wanting to just limit to complete empty string here?